src/Status/StatusIcon/StatusIcon.test.tsx

```diff
@@ -3,22 +3,20 @@ import * as React from "react";
 import statuses, { statusTypes } from "../statuses";
 
 import StatusIcon from "./StatusIcon";
+import { SvgIconComponent } from "@mui/icons-material";
 import { render } from "@testing-library/react";
 
+interface CustomIcon extends SvgIconComponent {
+  name?: string;
+}
+
 describe("StatusIcon", () => {
-  test.each(statusTypes)("renders correct color for %s", statusType => {
-    const { container } = render(<StatusIcon status={statusType} />);
-    // check that the avatar background color is correct
-    expect(container.querySelector(".MuiAvatar-root")).toHaveStyle({
-      backgroundColor: statuses[statusType].icon.color
-    });
-  });
   test.each(statusTypes)("renders correct icon for %s", statusType => {
     const { container } = render(<StatusIcon status={statusType} />);
     // confirm that the svg icon has the same name as we expect
     expect(container.querySelector("svg")).toHaveAttribute(
       "data-testid",
-      statuses[statusType].icon.type.name
+      (statuses[statusType].icon.type as CustomIcon).name
```

Comment:
This didn't look right to me so I did some digging. Turns out this test has never worked because this returned undefined, and toHaveAttribute apparently ignores the second input if undefined. This meant that all we actually tested was that the svg had a data-testid attribute, but didn't check what was in it.

To resolve this, I've remove all the manual typing that you did in the statuses.ts file. Typescript is able to infer all of this from the object definition, so we don't need to override any types.

Then in the tests, i've changed it to render the component, and the raw icon, and then check that the data-testids match. I've also documented exactly what is happening in the tests now as it was a but unclear before.